### DISCUSSION TITLE
how to effectively navigate between yaml and python files

### DISCUSSION THREAD

**Discussion by weiHelloWorld** [facebookresearch/hydra#2817] [Q&A]
hi, I am currently using Hydra for machine learning projects, and one issue I have is that I find it pretty inefficient to navigate between yaml and Python files, especially when there are multiple layers are yaml files.  For instance, suppose I have following yaml file:

```yaml
_target_: package_A.core.src.utils.func1.Model
...
```
how should I quickly jump from this file to Python file containing `package_A.core.src.utils.func1.Model`, especially if this yaml is not in the same folder as the corresponding python file?

Another scenario is navigating between different yaml files, for instance if I have this yaml:

```yaml
defaults:
    - data: myData
    - module: myModule
    - ....
```
where `myData`, `myModule` correspond to two yaml files.  How should I jump from this file to any of these two files?

Currently I mainly rely on search, but wonder if it is possible to navigate in these two scenarios more efficiently?

Thanks!

**Comment by odelalleau** [comment]
My guess is you'd need to write your own plugin for your IDE.
